src/macros.rs

```diff
@@ -140,87 +43,39 @@ macro_rules! for_match {
-///             .unwrap_or_else(|e|{
-///                 println!("{}",e);
-///                 std::process::exit(1)
-///             }) {
-///     println!("{} + 2: {}", v, v + 2);
-/// }
+/// println!("{} + 2: {}", len, len + 2);
 /// # }
 /// ```
 #[macro_export]
 macro_rules! value_t {
-    ($m:ident.value_of($v:expr), $t:ty) => {
-        match $m.value_of($v) {
-            Some(v) => {
-                match v.parse::<$t>() {
-                    Ok(val) => Ok(val),
-                    Err(_)  => Err(format!("'{}' isn't a valid value", ::clap::Format::Warning(v))),
-                }
-            },
-            None => Err(format!("The argument '{}' not found", ::clap::Format::Warning($v)))
-        }
+    ($m:ident, $v:expr, $t:ty) => {
+        value_t!($m.value_of($v), $t)
     };
-    ($m:ident.values_of($v:expr), $t:ty) => {
-        match $m.values_of($v) {
-            Some(ref v) => {
-                let mut tmp = Vec::with_capacity(v.len());
-                let mut err = None;
-                for pv in v {
-                    match pv.parse::<$t>() {
-                        Ok(rv) => tmp.push(rv),
-                        Err(e) => {
-                            err = Some(format!("'{}' isn't a valid value\n\t{}", ::clap::Format::Warning(pv),e));
-                            break
-                        }
-                    }
-                }
-                match err {
-                    Some(e) => Err(e),
-                    None => Ok(tmp)
-                }
-            },
-            None => Err(format!("The argument '{}' was not found", ::clap::Format::Warning($v)))
+    ($m:ident.value_of($v:expr), $t:ty) => {
+        if let Some(v) = $m.value_of($v) {
+            match v.parse::<$t>() {
+                Ok(val) => Ok(val),
+                Err(_)  =>
+                    Err(::clap::Error::value_validation(
+                        format!("The argument '{}' isn't a valid value", v))),
+            }
+        } else {
+            Err(::clap::Error::argument_not_found($v))
         }
     };
 }
 
-/// Convenience macro getting a typed value `T` where `T` implements `std::str::FromStr`
-/// This macro returns a `T` or `Vec<T>` or exits with a usage string upon failure. This
-/// removes some of the boiler plate to handle failures from value_t! above.
-///
-/// You can use it to get a single value `T`, or a `Vec<T>` with the `values_of()`
-///
-/// **NOTE:** This should only be used on required arguments, as it can be confusing to the user
-/// why they are getting error messages when it appears they're entering all required arguments.
+/// Convenience macro getting a typed value `T` where `T` implements `std::str::FromStr` or
+/// exiting upon error instead of returning a `Result`
 ///
-/// **NOTE:** Be cautious, as since this a macro invocation it's not exactly like
-/// standard syntax.
+/// **NOTE:** This macro is for backwards compatibility sake. Prefer
```

Comment:
I'm about 50/50 on this. There's nothing wrong with keeping it, I just think it's less "correct" but it is shorter so who knows. I say we keep it for now, and maybe deprecate at a later date.